packages/hub/process-controllers/hub-event-listener-controller.ts

```diff
@@ -0,0 +1,54 @@
+import { ContractSubscriptionEventHandler } from '../services/contract-subscription-event-handler';
+import { runInitializers, createRegistry } from '../main';
+import * as Sentry from '@sentry/node';
+import { Registry, Container } from '@cardstack/di';
+import { eventListenerLog } from '../utils/logger';
+
+export class HubEventListenerController {
+  logger = eventListenerLog;
+  static logger = eventListenerLog;
+
+  static async create(serverConfig?: {
+    registryCallback?: (r: Registry) => void;
+  }): Promise<HubEventListenerController> {
+    this.logger.info(`booting pid:${process.pid}`);
+    runInitializers();
+
+    let registry = createRegistry();
+    if (serverConfig?.registryCallback) {
+      serverConfig.registryCallback(registry);
+    }
+    let container = new Container(registry);
+    let handler: ContractSubscriptionEventHandler | undefined;
+
+    let contracts = await container.lookup('contracts');
+    let web3 = await container.lookup('web3-socket');
+    let workerClient = await container.lookup('worker-client');
+    let latestEventBlockQueries = await container.lookup('latest-event-block-queries');
+
+    try {
+      let handler = await container.instantiate(
+        ContractSubscriptionEventHandler,
+        web3,
+        workerClient,
+        contracts,
+        latestEventBlockQueries
+      );
+      await handler.setupContractEventSubscriptions();
```

Comment:
The reason we made `ContractSubscriptionEventHandler` not a service is that we were afraid it might be instantiated elsewhere, so making it not easily injectable was a way to hopefully prevent that. But it wasn’t really that sensible maybe. Now that you’ve made it a service, it can just use `inject` like other services, vs having the dependencies passed in via the constructor.

Reply:
Thanks for explaining - makes sense. Will adjust it.